plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityAnger.java

```diff
@@ -3,115 +3,56 @@
 import com.denizenscript.denizen.objects.EntityTag;
 import com.denizenscript.denizencore.objects.core.DurationTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.objects.properties.PropertyParser;
 import org.bukkit.entity.Bee;
-import org.bukkit.entity.Entity;
 import org.bukkit.entity.PigZombie;
 
-public class EntityAnger implements Property {
+public class EntityAnger extends EntityProperty<DurationTag> {
 
-    public static boolean describes(ObjectTag entity) {
-        if (!(entity instanceof EntityTag)) {
-            return false;
-        }
-        Entity bukkitEntity = ((EntityTag) entity).getBukkitEntity();
-        return bukkitEntity instanceof PigZombie
-                || bukkitEntity instanceof Bee;
+    // <--[property]
+    // @object EntityTag
+    // @name anger
+    // @input DurationTag
+    // @description
+    // Controls the remaining anger time of a PigZombie or Bee.
+    // -->
+
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof Bee
+                || entity.getBukkitEntity() instanceof PigZombie;
     }
 
-    public static EntityAnger getFrom(ObjectTag entity) {
-        if (!describes(entity)) {
-            return null;
+    @Override
+    public DurationTag getPropertyValue() {
+        if (getEntity() instanceof PigZombie pigZombie) {
+            return new DurationTag((long) pigZombie.getAnger());
         }
         else {
-            return new EntityAnger((EntityTag) entity);
+            return new DurationTag((long) as(Bee.class).getAnger());
         }
     }
 
-    public static final String[] handledMechs = new String[] {
-            "anger"
-    };
-
-    public EntityAnger(EntityTag entity) {
-        this.entity = entity;
-    }
-
-    EntityTag entity;
-
     @Override
-    public String getPropertyString() {
-        return new DurationTag((long) getAnger()).identify();
-    }
-
-    @Override
-    public String getPropertyId() {
-        return "anger";
-    }
-
-    public int getAnger() {
-        if (isPigZombie()) {
-            return getPigZombie().getAnger();
+    public void setPropertyValue(DurationTag param, Mechanism mechanism) {
+        if (mechanism.getValue().isInt()) { // Soft-deprecated - backwards compatibility, as this used to use a tick count
+            param = new DurationTag(mechanism.getValue().asLong());
         }
         else {
-            return getBee().getAnger();
+            param = mechanism.valueAsType(DurationTag.class);

```

Comment:
I believe this `else{}` block setter is redundant in the new `setPropertyValue` context

Please also test thoroughly to validate that the legacy tick input works as intended, since the code is used in a different context here